framework/components/ecl-navigation/ecl-navigation-menus/megamenu.js

```diff
@@ -1,24 +1,77 @@
+import { queryAll } from '@ec-europa/ecl-base/helpers/dom';
 import { toggleExpandable } from '@ec-europa/ecl-expandables/expandables';
 
-export const megamenu = selector => {
-  const megamenusArray = Array.prototype.slice.call(
-    document.querySelectorAll(selector)
-  );
+const onClick = (node, menu) => e => {
+  if (node && node.hasAttribute('aria-haspopup')) {
+    const hasPopup = node.getAttribute('aria-haspopup');
+    if (hasPopup === '' || hasPopup === 'true') {
+      e.preventDefault();
+
+      toggleExpandable(node, {
+        context: menu,
+        closeSiblings: true,
+      });
+    }
+  }
+};
+
+const onKeydown = (node, menu) => e => {
+  const currentTab = node.parentElement;
+  const previousTabItem =
+    currentTab.previousElementSibling ||
+    currentTab.parentElement.lastElementChild;
+  const nextTabItem =
+    currentTab.nextElementSibling || currentTab.parentElement.firstElementChild;
+
+  // don't catch key events when ⌘ or Alt modifier is present
+  if (e.metaKey || e.altKey) return;
+
+  // catch left/right and up/down arrow key events
+  // if new next/prev tab available, show it by passing tab anchor to showTab method
+  switch (e.keyCode) {
+    // ENTER or SPACE
+    case 13:
+    case 32:
+      onClick(e.currentTarget, menu)(e);
+      /* e.preventDefault();
+      toggleExpandable(e.currentTarget, {
+        context: menu,
+        closeSiblings: true,
+      }); */
+      break;
+    // ARROWS LEFT and UP
+    case 37:
+    case 38:
+      e.preventDefault();
+      previousTabItem.querySelector('a').focus();
+      break;
+    // ARROWS RIGHT and DOWN
+    case 39:
+    case 40:
+      e.preventDefault();
+      nextTabItem.querySelector('a').focus();
+      break;
+    default:
+      break;
+  }
+};
+
+export const megamenu = (
+  {
+    selector: selector = '.ecl-navigation-menu__root',
+    linkSelector: linkSelector = '.ecl-navigation-menu__link',
+  } = {}
+) => {
+  const megamenusArray = queryAll(selector);
 
   megamenusArray.forEach(menu => {
     // Get expandables within the menu
-    const nodesArray = Array.prototype.slice.call(
-      menu.querySelectorAll('[aria-controls][aria-expanded]')
-    );
-
-    nodesArray.forEach(node =>
-      node.addEventListener('click', () => {
-        toggleExpandable(node, {
-          context: menu,
-          closeSiblings: true,
-        });
-      })
-    );
+    const nodesArray = queryAll(linkSelector, menu);
+
+    nodesArray.forEach(node => {
+      node.addEventListener('click', onClick(node, menu));
```

Comment:
Not obligatory, rather nice to have for consistency:

```js
  // REVEAL API
  return {
    init,
    destroy,
  };
```

PS new the instantiation model is much better btw, thanks!